### PR TITLE
Fix dropdown backdrop

### LINKED ISSUE #697: dropdown should use the max width of it's longest content
As seen on the status dropdown, the text is truncated to "Under Rev...", the text should display the full content which is under "Under Review"
Screenshot from 2021-11-22 16-31-51

### PR DESCRIPTION
**Pull request by crux25**
This PR fixes issue #706 where if a dropdown is opened its backdrop covers other dropdowns on the page. 
Result being that the other dropdowns on the page are not clickable.
Also closes #697

### CODE REVIEW

File: ui/decredmaterial/dropdown.go [hunk 1]
```diff
@@ -204,43 +216,72 @@ func (d *DropDown) Layout(gtx C, dropPos int, reversePos bool) D {
...
-					d.Width = w + 10
-					return lay
-				})
-			})
+			return d.closedLayout(gtx, iLeft, iRight)
 		}),
 	)
 }
 
+// openeLayout computes dropdown layout when dropdown is opened.
```
Comment (Sirmorrison) [inline ui/decredmaterial/dropdown.go]:
```suggestion
// openedLayout computes dropdown layout when dropdown is opened.
```

File: ui/decredmaterial/theme.go [hunk 2]
```diff
@@ -221,6 +221,19 @@ func (t *Theme) closeAllDropdownMenus(group uint) {
 	}
 }
 
+// isOpenDropdownGroup interate over Dropdowns registered as a member
```
Comment (Sirmorrison) [inline ui/decredmaterial/theme.go]:
```suggestion
// isOpenDropdownGroup iterate over Dropdowns registered as a member
```

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
PR looks good just some little typo to fix.

also to help resolve #697 since this issue is about dropdown, the line of code to be modified is https://github.com/planetdecred/godcr/pull/719/files#diff-aaf10f22f02f78a1c31b0a39904ae196a4373ab5cbaf90b46d0358f295d5ebadL171-R199

original code

```layout.Rigid(func(gtx C) D {
			gtx.Constraints.Max.X = gtx.Px(unit.Dp(110))
			if d.revs {
				gtx.Constraints.Max.X = gtx.Px(unit.Dp(100))
			}
			gtx.Constraints.Min.X = gtx.Constraints.Max.X
			return layout.Inset{
				Right: unit.Dp(5),
				Left:  unit.Dp(5),
			}.Layout(gtx, func(gtx C) D {
				lbl := d.theme.Body2(item.Text)
				if !d.isOpen && len(item.Text) > 9 {
					lbl.Text = item.Text[:9] + "..."
				}
				return lbl.Layout(gtx)
			})
		}),
```

Suggestion

```layout.Rigid(func(gtx C) D {
			gtx.Constraints.Max.X = gtx.Px(unit.Dp(115))
			if d.revs {
				gtx.Constraints.Max.X = gtx.Px(unit.Dp(105))
			}
			gtx.Constraints.Min.X = gtx.Constraints.Max.X
			return layout.Inset{
				Right: unit.Dp(5),
				Left:  unit.Dp(5),
			}.Layout(gtx, func(gtx C) D {
				lbl := d.theme.Body2(item.Text)
				if !d.isOpen && len(item.Text) > 12 {
					lbl.Text = item.Text[:12] + "..."
				}
				return lbl.Layout(gtx)
			})
		}),
```

File: ui/decredmaterial/dropdown.go (hunk 1 above)
Comment (crux25) [reply ui/decredmaterial/dropdown.go]: Resolved.

File: ui/decredmaterial/theme.go (hunk 2 above)
Comment (crux25) [reply ui/decredmaterial/theme.go]: Resolved.

**Review by Sirmorrison — APPROVED** [review]

**Pull request merged**